test/e2e/diagnose.go

```diff
@@ -43,22 +46,20 @@ func DiagnoseMe(clients *test.Clients, logger *logging.BaseLogger) {
 func checkCurrentPodCount(clients *test.Clients, logger *logging.BaseLogger) {
 	revs, err := clients.ServingClient.Revisions.List(metav1.ListOptions{})
 	if err != nil {
-		logger.Errorf(fmt.Sprintf("could not check current pod count: %v", err))
+		logger.Errorf("Could not check current pod count: %v", err)
 		return
 	}
 	for _, r := range revs.Items {
 		deploymentName := r.Name + "-deployment"
 		dep, err := clients.KubeClient.Kube.AppsV1().Deployments(test.ServingNamespace).Get(deploymentName, metav1.GetOptions{})
 		if err != nil {
-			logger.Errorf(fmt.Sprintf("could not get deployment %v", deploymentName))
+			logger.Errorf("Could not get deployment %v", deploymentName)
 			continue
 		}
-		want := dep.Status.Replicas
-		have := dep.Status.ReadyReplicas
-		if have != want {
-			logger.Infof("deployment %v has %v pods. wants %v.", deploymentName, have, want)
+		if want, have := dep.Status.Replicas, dep.Status.ReadyReplicas; have != want {
+			logger.Infof("Deployment %v has %v pods. wants %v.", deploymentName, have, want)
 		} else {
-			logger.Infof("deployment %v has %v pods.", deploymentName, have)
+			logger.Infof("Deployment %v has %v pods.", deploymentName, have)
```

Comment:
nit: I actually find it easier to still have 'wants' here as it is not clear from this string alone that we have hit the number of desired pods.

Reply:
Yeah, I thought of that as well.